superset/connectors/connector_registry.py

```diff
@@ -99,6 +100,41 @@ def get_datasource_by_id(  # pylint: disable=too-many-arguments
                 pass
         raise NoResultFound(_("Datasource id not found: %(id)s", id=datasource_id))
 
+    @classmethod
+    def get_user_datasources(cls, session: Session) -> List["BaseDatasource"]:
+        from superset import security_manager
+
+        # collect datasources which the user has explicit permissions to
+        user_perms = security_manager.user_view_menu_names("datasource_access")
+        schema_perms = security_manager.user_view_menu_names("schema_access")
+        user_datasources = set()
+        for datasource_class in ConnectorRegistry.sources.values():
+            user_datasources.update(
+                session.query(datasource_class)
```

Comment:
@betodealmeida I believe this logic, i.e., relying on the fragile `perm` and `schema_perm` columns in the database, breaks if people are using a custom security manager. Should we be calling [get_datasources_accessible_by_user](https://github.com/apache/superset/blob/a06a2f30af02820189450b2264fa6bb734507ed8/superset/security/manager.py#L508) instead?

Note I sense currently Superset doesn't do a great job of differentiating between metadata and data access. The `get_datasources_accessible_by_user` is somewhat of a misnomer as it is merely used for metadata access (for an awareness perspective).

Reply:
@john-bodley, `get_datasources_accessible_by_user` calls `ConnectorRegistry.query_datasources_by_permissions`, which uses the same logic I'm using here.

Since we're getting `user_perms` and `schema_perms` from the security manager (lines 108, 109) doesn't it means this works with custom security managers?

Reply:
@betodealmeida @john-bodley I think that the question here is that AirBnb totally overrides the DB permissions, they do that by overriding the security manager, their permission "backend" is something totally different. So that's possible if all permission checks are done on the security manager. 

Makes me think how can that work on the REST API defined filters

Reply:
@dpgaspar that's correct. We actually don't use any of the FAB logic and thus we completely bypass the `securiy_manager.user_view_menu_names` and reliance of  the `datasource_class.perm` and `datasource_class.schema_perm` database columns.

Reply:
@betodealmeida if the `get_datasources_accessible_by_user` method isn't ideal, then I think the alternatively would be to move the `get_user_datasources` method to the security manager so deployments can overwrite the logic if necessary.